src/Microsoft.DncEng.SecretManager/StorageUtils.cs

```diff
@@ -49,6 +49,45 @@ public static async Task<string> RotateStorageAccountKey(string subscriptionId,
             return key.Value;
         }
 
+        public static SharedAccessAccountPermissions AccountPermissionsFromString(string input)
+        {
+            var accessAccountPermissions = SharedAccessAccountPermissions.None;
+            foreach (char ch in input)
+            {
+                accessAccountPermissions |= ch switch
+                {
+                    'a' => SharedAccessAccountPermissions.Add,
+                    'c' => SharedAccessAccountPermissions.Create,
+                    'd' => SharedAccessAccountPermissions.Delete,
+                    'l' => SharedAccessAccountPermissions.List,
+                    'r' => SharedAccessAccountPermissions.Read,
+                    'w' => SharedAccessAccountPermissions.Write,
+                    'u' => SharedAccessAccountPermissions.Update,
+                    'p' => SharedAccessAccountPermissions.ProcessMessages,
+                    _ => throw new ArgumentOutOfRangeException(nameof(input))
+                };
+            }
+
+            if (accessAccountPermissions == SharedAccessAccountPermissions.None)
```

Comment:
Wait... what? If it's exactly equal to this thing... then `|=` that exact thing? That... should be a no-op yes? What is this if statement attempting to do?

Reply:
This whole function is just a copypasta of the existing SharedAccessBlobPolicy.PermissionsFromString method to do account permissions. For some reason they didn't write this one too.

Reply:
Weird. This if _still_ doesn't do anything.  I don't particularly like this method, since it's got some... odd... behavior... and the single letters are a bit cryptic...  But if we want to leave it untouched because it's a copy of something, a reference to where the copy came from (in case they add/change anything there) would be good. :-)

Reply:
I'll clean this guy up, its not technically a copy, more a copy+edit a bit.